large_image/tilesource/geo.py

```diff
@@ -0,0 +1,519 @@
+from urllib.parse import urlencode, urlparse
+
+import pyproj  # TODO: import issues
+
+from large_image.cache_util import CacheProperties, methodcache
+from large_image.constants import TileInputUnits
+from large_image.exceptions import TileSourceError
+
+from .base import FileTileSource
+from .utilities import JSONDict, getPaletteColors
+
+TileInputUnits['projection'] = 'projection'
+TileInputUnits['proj'] = 'projection'
+TileInputUnits['wgs84'] = 'proj4:EPSG:4326'
+TileInputUnits['4326'] = 'proj4:EPSG:4326'
+
+# Inform the tile source cache about the potential size of this tile source
+CacheProperties['tilesource']['itemExpectedSize'] = max(
+    CacheProperties['tilesource']['itemExpectedSize'],
+    100 * 1024 ** 2)
+
+# Used to cache pixel size for projections
+ProjUnitsAcrossLevel0 = {}
+ProjUnitsAcrossLevel0_MaxSize = 100
+
+InitPrefix = '+init='
+NeededInitPrefix = '' if int(pyproj.proj_version_str.split('.')[0]) >= 6 else InitPrefix
+
+
+def make_vsi(url: str, **options):
+    if str(url).startswith('s3://'):
+        s3_path = url.replace('s3://', '')
+        vsi = f'/vsis3/{s3_path}'
+    else:
+        gdal_options = {
+            'url': str(url),
+            'use_head': 'no',
+            'list_dir': 'no',
+        }
+        gdal_options.update(options)
+        vsi = f'/vsicurl?{urlencode(gdal_options)}'
+    return vsi
+
+
+class GeoBaseFileTileSource(FileTileSource):
+    """Abstract base class for geospatial tile sources."""
+
+    geospatial = True
```

Comment:
We want this to be ` _geospatial_source = True`.  Remember, we changed the internal flag name so that the `geospatial` property was if the opened source had geospatial information.